libeos-updater-util/avahi-service-file.c

```diff
@@ -99,8 +99,8 @@ handle_text_record (GString      *records_str,
    */
   g_assert (total_size != NULL);
   *total_size += 1 + record_size;
-  escaped_key = g_markup_escape_text (key, key_length);
-  escaped_value = g_markup_escape_text (value_string, value_string_length);
+  escaped_key = g_markup_escape_text (key, (gssize) key_length);
```

Comment:
It might be worth bounds-checking `key_length` and `value_string_length` here as they come from an uncontrolled input.

Reply:
They’re checked as part of `record_size` by the `check_record_size()` call above.